src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageMetadataControl.xaml.cs

```diff
@@ -1,60 +1,98 @@
         {
-            if (DataContext is DetailedPackageMetadata metadata)
+            if (ReadMePreviewViewModel is not null
+                && e.PropertyName == nameof(ReadMePreviewViewModel.CanDetermineReadMeDefined))
             {
-                var window = new LicenseFileWindow()
+                SetReadmeTabVisibility(ReadMePreviewViewModel.CanDetermineReadMeDefined ? Visibility.Visible : Visibility.Collapsed);
+                if (!ReadMePreviewViewModel.CanDetermineReadMeDefined)
                 {
-                    DataContext = new LicenseFileData
-                    {
-                        LicenseHeader = string.Format(CultureInfo.CurrentCulture, UI.Resources.WindowTitle_LicenseFileWindow, metadata.Id),
-                        LicenseText = new FlowDocument(new Paragraph(new Run(UI.Resources.LicenseFile_Loading)))
-                    }
-                };
+                    SelectTab(PackageMetadataTab.PackageDetails);
+                }
+            }
+        }
 
-                NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
-                {
-                    string content = await PackageLicenseUtilities.GetEmbeddedLicenseAsync(new Packaging.Core.PackageIdentity(metadata.Id, metadata.Version), CancellationToken.None);
+        public ReadMePreviewViewModel ReadMePreviewViewModel { get => _readMePreviewViewModel; set => _readMePreviewViewModel = value; }
 
-                    var flowDoc = new FlowDocument();
-                    flowDoc.Blocks.AddRange(PackageLicenseUtilities.GenerateParagraphs(content));
-                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-                    (window.DataContext as LicenseFileData).LicenseText = flowDoc;
-                }).PostOnFailure(nameof(PackageMetadataControl), nameof(ViewLicense_Click));
+        public void SelectTab(PackageMetadataTab selectedTab)
+        {
+            switch (selectedTab)
+            {
+                case PackageMetadataTab.PackageDetails:
+                    SelectedTabItem = tabPackageDetails;
+                    break;
+                case PackageMetadataTab.Readme:
+                default:
+                    SelectedTabItem = tabReadMe;
+                    break;
+            }
+        }
 
-                window.ShowModal();
+        public void SetReadmeTabVisibility(Visibility visibility)
+        {
+            tabReadMe.Visibility = visibility;
+        }
+
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _packageMetadataReadMeControl?.Dispose();
+                ReadMePreviewViewModel.PropertyChanged -= ReadMePreviewViewModel_PropertyChanged;
```

Comment:
The guidance suggests setting `_disposed = true;` in this method.

https://learn.microsoft.com/dotnet/standard/garbage-collection/implementing-dispose#the-disposebool-method-overload